### ISSUE TITLE
Admin Interface: Playback Controls for Soheil's New Task Playback Tool

### ISSUE THREAD

**Issue by jonfroehlich** [ProjectSidewalk/SidewalkWebpage#391]
Would be nice to have a timeline and playback controls for @Soben713's new task playback tool. The timeline should have little indicators where activities occurred and you should be able to use a slider-bar (or similar) to slide back and forth across time.

**Comment by Soben713** [comment]
I agree that a timeline of activities and events would be very useful for a better understanding of the tasks. In fact, I also found myself need it as well and will implement it.

However, being able to go back and forth in time is not easy to implement at all (unless I find a library for D3 that does that). Note that there is only one street segment in every task and in most cases, tasks are finished in less than ~10 seconds. I personally refresh the page anytime I want to look at the task again.

**Comment by jonfroehlich** [comment]
I don't understand implementation difficulty. We can chat about it and decide if worth it. IMO, should be straightforward.

**Comment by jonfroehlich** [comment]
I still don't understand how/why this is difficult to implement. @Soben713 never answered this question. @sagec108, since you've been working on fixing this, perhaps you can shed light on why this is difficult.

**Comment by sagec108** [comment]
@jonfroehlich I touched on the difficulty in pull request #634. To restate it here, there are existing players for handling videos (resembling video players like you see on YouTube). However, in our case, since we are getting image by image, we need to create one from scratch or find a library that can address getting the specific image at each time point. After looking around for a while, I was not able to find a good solution that would handle the user pressing the slider handle. The option that showed the most promise to me was the bootstrap slider library, which I could animate in my own "try" file but I could not successfully integrate it into the website.

The current playback page uses a hardcoded html/css object as a makeshift slider, which holds no interactive use and relies on the jquery "animate" function. To have playback functionality, we need to be able to pause and restart animation at a whim, which means we can no longer rely on this function.

**Comment by jonfroehlich** [comment]
You haven't convinced me that this is a hard problem. :) I don't think we *should* use an existing video player for this. This seems like going overboard for a relatively simple interactive widget.

Happy to talk to you in person (higher bandwidth) and you can then synthesize our discussion back to this github issue.

**Comment by jonfroehlich** [comment]
Closing now. See reasoning: https://github.com/ProjectSidewalk/SidewalkWebpage/issues/490
